Formula/makensis.rb

```diff
@@ -23,12 +25,15 @@ def install
     args = [
       "CC=#{ENV.cc}",
       "CXX=#{ENV.cxx}",
+      "PREFIX=#{prefix}",
       "PREFIX_DOC=#{share}/nsis/Docs",
       "SKIPUTILS=Makensisw,NSIS Menu,zip2exe",
       # Don't strip, see https://github.com/Homebrew/homebrew/issues/28718
       "STRIP=0",
       "VERSION=#{version}",
     ]
+    on_linux { args << "APPEND_LINKFLAGS=-Wl,-rpath,#{rpath}" }
```

Comment:
or maybe:
```suggestion
    on_linux { args << "APPEND_LINKFLAGS=-Wl,-rpath=#{Formula["zlib"].opt_lib}" }
```

Reply:
I'd rather we use specific formulae whenever possible.  If a user were to unlink `zlib` (not that anyone should do this), then `libz.so` would not be found in `$HOMEBREW_PREFIX/lib`.  However when using the `opt_lib`, the only way this could ever be deleted is if `zlib` were uninstalled and that would require them to use `--force`, which is not the default option.

Reply:
That is fine with me. I don't think any of the other dynamic links need Homebrew's path.

As a record, this is how current bottle looks:
```
Dynamic Section:
  NEEDED       libpthread.so.0
  NEEDED       libz.so.1
  NEEDED       libstdc++.so.6
  NEEDED       libgcc_s.so.1
  NEEDED       libc.so.6
  RPATH        @@HOMEBREW_PREFIX@@/lib
```

Previous attempt at bottling kept missing `RPATH`, though I have no idea why.

---

On side note, when I built in a local docker container (with various other Homebrew packages installed), it automatically created `RPATH` with gcc-5 path and Homebrew's zlib path.

However, when I tried from clean Homebrew Docker, it ended up the same as CI run without any `RPATH`.

Reply:
Was the local container using brewed `gcc@5`?  If so, the specs file we ship with it adds the RPATH for gcc-5 to anything it builds.  However, the Homebrew Docker does not use brewed `gcc@5` to build anything, but is instead using the host gcc-5 which does not add an RPATH for itself because it is in `/usr`, which the linker will always find.  The fact that we don't add the RPATH for brewed `gcc@5` has created some problems in the past, although we made some changes that fixed this recently: https://github.com/Homebrew/brew/pull/11459.

Reply:
That might have been the reason, in which case, it is possible that the RPATH logic in this formula for adding zlib was dependent on an existing value.

For example, the zlib dir was added to RPATH when `gcc@5` was already on the RPATH.

The PR was merged with existing general Homebrew lib dir. I guess we can keep it for now until either a version bump or if someone actually tries to unlink `zlib` and opens an issue.